clients/mobile/src/components/button/SocialButton.tsx

```diff
@@ -1,24 +1,27 @@
 import React from 'react'
-import { TouchableOpacity, Text, StyleSheet, Image, View, ImageSourcePropType, StyleProp, ViewStyle, TextStyle } from 'react-native'
+import { TouchableOpacity, Text, StyleSheet, Image, View, ImageSourcePropType, StyleProp, ViewStyle, TextStyle, ActivityIndicator } from 'react-native'
 import { useTheme } from '../../setup/theme/hooks/useTheme'
 import { Theme } from '../../setup/theme'
 
 interface SocialButtonProps {
   text: string;
   onPress: () => void;
   icon: ImageSourcePropType;
+  loading?: boolean;
   buttonStyle?: StyleProp<ViewStyle>;
   textStyle?: StyleProp<TextStyle>;
 }
 
-export const SocialButton = ({ text, onPress, icon, buttonStyle, textStyle }: SocialButtonProps) => {
-  const styles = createStyles(useTheme())
+export const SocialButton = ({ text, onPress, icon, loading, buttonStyle, textStyle }: SocialButtonProps) => {
+  const theme = useTheme()
+  const styles = createStyles(theme)
+  const isLoading = Boolean(loading)
```

Comment:
Above you define the type for `loading` as `boolean` but again you are converting loading to boolean why?

Reply:
good catch vai, updated [I-](https://github.com/CraftsmenLtd/BloodConnect/commit/05aa9940c46d98434d838056c59e651d6b97231e)https://github.com/CraftsmenLtd/BloodConnect/issues/231[: Remove redundant type conversion since loading prop is already boolean](https://github.com/CraftsmenLtd/BloodConnect/commit/05aa9940c46d98434d838056c59e651d6b97231e)